### PR TITLE
Better KCFG visualization

### PR DESCRIPTION
**Pull request by ehildenb**
This PR makes some improvements to the KCFG visualization, after discussions with the textualize developers in [their discord](https://discord.gg/qPJkUrxc).

- The `KCFG.pretty` is factored into functionality `KCFG.pretty_segments`, which returns each _chunk_ of lines separately with an identifier attached.
- The `KCFG.pretty` outputs more compact textual representation, which is a bit more visually appealing.
- The `KCFGViewer` has better CSS styling/presentation.
- The `KCFGViewer` is more modular, instead of being one monolithic application.
- You can now click on nodes in the KCFG viewer, and it will jump you straight to that node.

See https://drive.google.com/file/d/1ThtFn8hcTc3P5lsviz4DR2R6mZyJjCxc/view for a video demo of new functionality.

### CODE REVIEW

**Review by radumereuta — APPROVED** [review]
lgtm but my confidence is sort of low on this one.
It's a lot of new code to me.
One suggestion I have is to change `shorten_hashes` to pick the first seven digits. GitHub does this. I find it more readable this way and takes less space.

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
 name = "pyk"
-version = "0.1.91"
+version = "0.1.92"
```
Comment (tothtamas28) [inline pyproject.toml:7]: `kcfg-viewer` should be removed from `tool.poetry.scripts`.

File: src/pyk/kcfg/kcfg.py [hunk 2]
```diff
@@ -344,11 +344,12 @@ def node_short_info(self, node: Node, node_printer: Optional[Callable[[CTerm], I
             node_strs.extend(f' {nl}' for nl in node_printer(node.cterm))
         return node_strs
 
-    def pretty(
+    def pretty_segments(
```
Comment (tothtamas28) [inline src/pyk/kcfg/kcfg.py:347]: Please add docstrings for how the return value should be interpreted.

File: src/pyk/kcfg/kcfg.py [hunk 3]
```diff
@@ -362,76 +363,100 @@ def _print_node(node: KCFG.Node) -> List[str]:
...
+            _ret_lines.append((f'{new_id}', [l.rstrip() for l in seg_lines]))
+        return _ret_lines
+
+    def pretty(
+        self, kprint: KPrint, minimize: bool = True, node_printer: Optional[Callable[[CTerm], Iterable[str]]] = None
+    ) -> Iterable[str]:
+        return (
+            l
+            for _, seg_lines in self.pretty_segments(kprint, minimize=minimize, node_printer=node_printer)
+            for l in seg_lines
```
Comment (tothtamas28) [inline src/pyk/kcfg/kcfg.py:458]:
```suggestion
            line
            for _, seg_lines in self.pretty_segments(kprint, minimize=minimize, node_printer=node_printer)
            for line in seg_lines
```

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**